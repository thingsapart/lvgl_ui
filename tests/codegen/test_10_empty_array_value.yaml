# Tests a key with an explicit empty array `[]` as its value.
- type: button
  pos: [10, 10]
  size: [120, 50]
  remove_style_all: []
  children:
    - type: label
      text: A label inside button w/o styles
