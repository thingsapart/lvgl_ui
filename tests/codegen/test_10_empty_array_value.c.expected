/* AUTO-GENERATED by the 'c_code' backend */

#include "lvgl.h"
#include "c_gen/lvgl_dispatch.h" // For obj_registry_add
#include "data_binding.h"

void create_ui(lv_obj_t* parent) {
    // unnamed: button_0 (button)
    lv_obj_t* button_0 = lv_button_create(parent);

    lv_obj_set_pos(button_0, 10, 10);
    lv_obj_set_style_size(button_0, 120, 50, 0);
    lv_obj_remove_style_all(button_0);
    // unnamed: label_1 (label)
    lv_obj_t* label_1 = lv_label_create(button_0);

    lv_label_set_text(label_1, "A label inside button w/o styles");


}
